src/ipc/ipc4/dai.c

```diff
@@ -115,6 +121,15 @@ int dai_config_dma_channel(struct dai_data *dd, struct comp_dev *dev, const void
 		 */
 		channel = 0;
 		break;
+	case SOF_DAI_INTEL_UAOL:
+#if defined(CONFIG_ACE_VERSION_2_0) || defined(CONFIG_ACE_VERSION_3_0)
+		channel = 0;
+		if (dai->host_dma_config[0]->pre_allocated_by_host)
+			channel = dai->host_dma_config[0]->dma_channel_id;
+#else
+		channel = copier_cfg->gtw_cfg.node_id.f.v_index;
+#endif
+		break;
```

Comment:
I don't fully understand this.

for LNL+ all the DMA stuff is handled in the same way with an HDaudio-based solution.

We just moved some of the ALH stuff into the HDaudio prodessing - see line 94 the test for ALH.

so shouldn't the UAOL stuff also be moved under the HDAudio category? Why special case what looks identical?

Reply:
my comment still stands.

Reply:
Removed support for MTL, so the line is no longer present.
However the concept of handling UAOL by HDA code (and removing of all UAOL related stuff from SOF) seems not to be applicable. This is because of UAOL being a type of DAI handled by separate Zephyr driver with the specific API.